### PR TITLE
test(platform-server): setup framework for platform-server e2e tests …

### PR DESCRIPTION
**Pull request by vikerman**
…and add a simple helloworld test

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ X] Other... Please describe: Add e2e tests for platform-server
```

### CODE REVIEW

File: packages/platform-server/integrationtest/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/rxjs/
+built/
+dist/
```
Comment (alxhub) [inline packages/platform-server/integrationtest/.gitignore]: `/dist` unless you intend to ignore any directory `dist/` anywhere in the hierarchy.

File: packages/platform-server/integrationtest/e2e/helloworld-spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    // Test the contents from the server.
+    const serverDiv = browser.driver.findElement(by.css('div'));
+    expect(serverDiv.getText()).toEqual('Hello world!');
+
+    // Bootstrap the client side app.
+    browser.executeScript('doBootstrap()');
+
+    // Retest the contents after the client bootstraps.
+    const div = element(by.css('div'));
+    expect(div.getText()).toEqual('Hello world!');
```
Comment (alxhub) [inline packages/platform-server/integrationtest/e2e/helloworld-spec.ts]: Maybe use `{{PLATFORM_NAME}}` in the root component's template?

**Review by alxhub — APPROVED** [review]

File: packages/platform-server/integrationtest/build.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/build.sh]: add `-u`

File: packages/platform-server/integrationtest/run_tests.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/run_tests.sh]: add `-u`

File: packages/platform-server/integrationtest/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  },
+
+  "exclude": [
+    "vendor",
+    "node_modules",
+    "built",
+    "dist",
+    "e2e"
+  ]
+}
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/tsconfig.json]: add new line

File: packages/platform-server/integrationtest/src/helloworld/app.server.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+import {HelloWorldModule} from './app';
+import {HelloWorldComponent} from './hello-world.component';
+
+@NgModule({
+  bootstrap: [HelloWorldComponent],
+  imports: [HelloWorldModule, ServerModule],
+})
+export class HelloWorldServerModule {
+}
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/src/helloworld/app.server.ts]: add new line

File: packages/platform-server/integrationtest/.gitignore [hunk 7]
```diff
@@ -0,0 +1,7 @@
+/rxjs/
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/.gitignore]: why `/` prefix? makes it look like an absolute path

File: scripts/ci/test-e2e.sh [hunk 8]
```diff
@@ -29,6 +29,9 @@ travisFoldStart "test.e2e.offlineCompiler"
   ${thisDir}/offline_compiler_test.sh
 travisFoldEnd "test.e2e.offlineCompiler"
 
+travisFoldStart "test.e2e.platform-server"
+  ./packages/platform-server/integrationtest/run_tests.sh
```
Comment (IgorMinar) [inline scripts/ci/test-e2e.sh]: the `e2e` job is already taking too long. Can you please setup a new Travis job and put this test there?

you could also move a few other checks into this new job to decrease the time it takes to run the e2e job.

**Review by IgorMinar — CHANGES_REQUESTED** [review]

File: packages/platform-server/integrationtest/.gitignore [hunk 9]
```diff
@@ -0,0 +1,6 @@
+built/
```
Comment (IgorMinar) [inline packages/platform-server/integrationtest/.gitignore]: is "built" used anywhere else? we usually use "dist"

**Review by IgorMinar — APPROVED** [review]
This change now shaves off 5 minutes from our overall CI run. Good job @vikerman!!

**Pull request merged**